docs/docs/spec.md

```diff
@@ -927,36 +927,28 @@ the output of the following operation:
     > demonstrated in
     > https://github.com/jspecify/jspecify-reference-checker/pull/197.
 
-    > The purpose of this special case is to improve behavior in "the
-    > `ImmutableList.Builder` case": Because `ImmutableList.Builder.add` always
-    > throws `NullPointerException` for a null argument, we would like for
-    > `add(null)` to be a compile error, even under lenient tools.
-    > Unfortunately, without this special case, lenient tools could permit
-    > `add(null)` in unannotated code. For an example, read on.
+    > The purpose of this part of the subsitution rules is ensure that non-null
+    > types stay non-null during substitution, even if they don't have an
+    > explicit `@NonNull` annotation on them.
     >
-    > Consider an unannotated user of `ImmutableList.Builder<Foo> builder`. Its
-    > type argument `Foo` will have a [nullness operator] of `UNSPECIFIED`.
-    > Without this special case, the parameter of `builder.add` would have a
-    > nullness operator of `UNSPECIFIED`, too. Then, when a lenient tool would
-    > check whether the [some-world] subtyping relation holds for
-    > `builder.add(null)`, it would find that it does.
+    > For an example of such a type, consider `Comparable`, a `@NullMarked`
+    > interface that declares a non-nullable type parameter `T` and a method
+    > `compare(T)`. By JSpecify rules, the method parameter has type `T
+    > NO_CHANGE`, and that type is null-exclusive under every parameterization
+    > in all worlds. Now consider a null-unmarked class that declares a method
+    > `Comparable<Foo> foo()`, which by JSpecify rules has a type argument `Foo
```

Comment:
Is it worth having `@NonNull Comparable<Foo>` here, to avoid questions about dereferencing `foo()` below? Or is that too distracting, and people will know what you mean?